test/rules/no-restricted-patterns/no-restriced-patterns.js

```diff
@@ -1,66 +1,200 @@
 var ruleTestBase = require('../rule-test-base');
 var rule = require('../../../dist/rules/no-restricted-patterns.js');
-var runTest = ruleTestBase.createRuleTest(rule, 'Restricted pattern "<%= steps %>" on <%= nodeType %>');
+var runTest = ruleTestBase.createRuleTest(rule, '<%= nodeType %> <%= property %>: "<%= string %>" matches restricted pattern "/<%= pattern %>/i"');
 
-describe('No Allowed Steps Rule', function() {
+describe('No Restricted Patterns Rule', function() {
   it('doesn\'t raise errors when there are no violations', function() {
     runTest('no-restricted-patterns/NoViolations.feature', {
       'Global': [
         '^.*disallowed.*$'
       ]}, []);
   });
 
-  it('detects errors for features, scenarios, backgrounds, and scenario outlines', function() {
-    runTest('no-restricted-patterns/Violations.feature', {
+  it('detects errors in Feature names and descriptions that match the Feature or Global config', function() {
```

Comment:
I split up the tests, because the number of errors was becoming pretty big and the file was becoming unreadable.